### ISSUE TITLE
Identificador único padronizado dos equipamentos (espaço) diferente do id de sistema.

### ISSUE THREAD

**Issue by intrepidocabral** [culturagovbr/mapasculturais_descontinuado#47]
GT de ontologia terminara trabalho em dezembro. Para viabilidade futura de identificador unico em consonancia com glossario da cultura, inserir id-unico: composto com id com tipologia, dado ibge, id

**Comment by marylly** [comment]
Após conferência sobre o assunto realizada dia 18/01/2017, foram feitas algumas propostas de identificação única de equipamentos na plataforma Mapas Culturais:
**Agente:**
- Estado+Município+Localização+Entidade
- Estado+Município+Localização+Entidade+Documento
- Estado+Município+Localização+Entidade+Nome
- Entidade+Estado+Município+Nome+Localização

**Espaço:**
- Estado+Munícipio+Localização+Entidade
- Entidade+Estado+Munícipio+Localização

**Comment by marylly** [comment]
Precisamos fechar esse item para dimensionar e definir uma data de conclusão @thaisrigolon @GustavoVS @robertkapa

**Comment by thaisrigolon** [comment]
@marylly quer marcar um hangout pra fecharmos esse item?

**Comment by marylly** [comment]
@thaisrigolon pode ser na segunda,depois da reunião com o MinC?

**Comment by thaisrigolon** [comment]
Sim, @marylly. Você avisa na comunidade de desenvolvedores pra quem quiser/puder também participar?

**Comment by marylly** [comment]
**Agente:**
Entidade+Estado+Município+Localização+Nome

**Espaço:**
Entidade+Estado+Munícipio+Localização+Nome

Onde é um princípio de chave única para as entidades principais agente e espaço para a plataforma do mapas.

As informações de localização podemos pensar em algum algoritmo que encurte a informação.

Talvez o Nome das entidades pode ser armazenado sem acentos.

**Comment by leogermani** [comment]
Documentando o q conversei com a Marylly, acho que deveria acrescentar o nome da entidade,

pq nesse caso aí em cima um espaço e um subespaço no mesmo endereço teriam o mesmo id.

e acho que a localização poderia ser cortando um ou dois dígitos das coordenadas, assim coisas que estão muito pertos teriam o mesmo id.

Mas assim: precisqa deicidir pra que queremos esse ID. Estou imaginando algo que sirva apenas para sinalizar possíveis duplicações...

Indo mais adiante, não vejo pq implementar isso agora se não vamos usar pra nada. Acho que isso poderia ser feito quando formos desenvolver alguma ferramenta pra achar duplicidades, ou quando formos de fato integrar instalações distintas...

Fazer isso agora e só ir salvando no banco, sem ter claro como vai ser usado, não adianta nada... E quando decidirmos como usar, saberemos o melhor formato, e aí é só fazer um script pra gerar pra todo mundo...

**Comment by avaloni** [comment]
Isso aqui é uma chave única nas entidades Agente e Espaço(por enquanto), certo? 

Nesse caso, o que faríamos com as duplicações que existem nas instalações atuais? Inserir a chave composta única não seria possível sem remover os duplicados.

**Comment by thaisrigolon** [comment]
> Isso aqui é uma chave única nas entidades Agente e Espaço(por enquanto), certo?
> 
> Nesse caso, o que faríamos com as duplicações que existem nas instalações atuais? Inserir a chave composta única não seria possível sem remover os duplicados.

A chave única é pra identificar os dados duplicados.

**Exemplo 1:** ao integrar o banco de um município e do estado seria bem fácil de descobrir as duplicações.
Com os dados atuais, facilitaria a moderação no painel para o admin. Seria legal apresentar os dados duplicados. 
**Exemplo 2:** Quando entrar num espaço informar qual é o outro espaço que está duplicado. O administrador teria a possibilidade de excluir o duplicado e/ou passar a propriedade para o agente correto.

**Comment by diegodorgam** [comment]
ok então, só registrando as evoluções dessa issue que tivemos por email:
@avaloni pode por gentileza implementar os campos de identificação seguindo a lógica:

> Agente:
> Entidade+Estado+Município+Localização+Nome
> Espaço:
> Entidade+Estado+Munícipio+Localização+Nome

**Comment by diegodorgam** [comment]
@thaisrigolon @marcus2703 a @marylly referencia nessa issue o commit com a funcionalidade de id unico. 

já está funcionando. 

Quando quiserem testar..

Precisa baixar esse commit no repositório e rodar o db_updates.sh
a partir daí, depois que uma entidade é criada, ela cria mais um campo nas tabelas agent e space, chamado `unique_id`.

**Comment by leogermani** [comment]
Pessoal, meus comentários:

1. Se isso ainda está para ser testado, deveria ter continuado na branch de feature, e não ter sido feito o merge pro develop.

2. Como a gente já conversou há um tempão atrás, não tem sentido a gente implementar um campo que a gente não vai usar agora. Quando a gente for usar, a gente implementa e aí roda o db update. A gente não ganha nada fazendo isso agora, só mais overhead, mais código pra manter, mais coluna no banco de dados, mais deploy complicado pq tem um db update gigante pra rodar...

Sobre o esquema de id que foi feito, eu até teria comentários. Tipo, acho que podia pegar só um pedaço da latitudo e longitude, excluindo um ou dois dígitos finais, pq aí poderia acusar pontos que ficaram muito próximos, mas não exatamente no mesmo lugar.

Mas isso pouco importa. Sem dúvida na hora que formos de fato fazer uma busca por itens duplicados vamos poder testar e validar o melhor formato...

Ou vocês chegaram a fazer isso com a base do MinC?

Enfim, minha opinião. Estou inclinado a remover isso da branch develop.

**Comment by leogermani** [comment]
mais um comentário, olhar http://docs.mapasculturais.org/mc_developer_contribute/ antes de mandar qq coisa pro develop

**Comment by leogermani** [comment]
Pessoal,

Testei aqui e está funcionando. Vou reorganizar as branches pra isso ficar no lugar certinho. Por mim podemos considerar essa tarefa concluída.
